spark/src/test/scala/geotrellis/spark/io/avro/AvroTools.scala

```diff
@@ -1,14 +1,172 @@
 package geotrellis.spark.io.avro
 
 import org.scalatest._
+import Matchers._
+
+import geotrellis.raster._
 
 trait AvroTools { self: Matchers =>
+  import AvroTools._
+
   def roundTrip[T](thing: T)(implicit codec: AvroRecordCodec[T]): Unit = {
     val bytes = AvroEncoder.toBinary(thing)
     val fromBytes = AvroEncoder.fromBinary[T](bytes)
-    fromBytes should be equals thing
+    fromBytes shouldBe thing
     val json = AvroEncoder.toJson(thing)
     val fromJson = AvroEncoder.fromJson[T](json)
-    fromJson should be equals thing
+    fromJson shouldBe thing
+  }
+
+  def roundTripWithNoDataCheck[T <% NoDataValueChecker[T]](thing: T)(implicit codec: AvroRecordCodec[T]): Unit = {
```

Comment:
Viewable is actually deprecated (or being deprecated) in Scala, so we avoid using it (`<%`)

A pattern I would use here is:

```scala
import geotrellis.util.MethodExtensions

trait AvroTools {
  def roundTripWithNoDataCheck[T: AvroRecordCodec: (? => AvroNoDataCheckMethods[T])](thing: T): Unit = [
    // Now we can call those methods on thing
  }
}

object AvroTools {
  trait NoDataValueChecker[T] extends MethodExtensions[T] { ... }
  implicit class ShortNoDataValueChecker(val self: ShortArrayTile) extends NoDavaValueChecker[ShortArrayTile] {
    override def doCheck(nodata: Option[Any]): Unit = { ... }
  }
}
```

This would fit the method extension structure of the rest of the library.